### DISCUSSION TITLE
How to check that constructor's arg have same order as declared fields

### DISCUSSION THREAD

**Discussion by snuyanzin** [checkstyle/checkstyle#14994] [Q&A]
e.g.
a class
```java
public class A {
   private int a;
   private String b;
   private long c;
   public A (long c, String b, int a) { // wrong since args ordered by type don't match order of fields by type
      this.b = b;    //wrong since order of initialization doesn't match order of fields, probably forward reference should be skipped  
      this.a = a;
      this.c = c;
   }
...
}
```
There could be 2 types of checks as mentioned in comments
I tried `DeclarationOrder` (https://checkstyle.sourceforge.io/checks/coding/declarationorder.html)
and it seems it doesn't cover this case

Would be great to hear more insights how to make this kind of check possible

**Comment by rnveach** [comment]
Checkstyle doesn't do this kind of check. `DeclarationOrder` just makes sure items (fields,methods, etc...) follow a certain order in the class, not in the constructor. I had to implement my own version to get this type of validation. You can try requesting a check like this by raising an issue.

**Reply by romani** [reply]
@snuyanzin 
you need to write your own Check , https://checkstyle.org/writingchecks.html
we have numerous examples in our code base how to do.
Here is example of thirdparty (not standard Checks) - https://github.com/sevntu-checkstyle/sevntu.checkstyle/tree/master/sevntu-checks
that exists on a side.
Just do your own Check and it will control code to be as you want it to be.

Keep in mind that Constructor is not always receive same types as it has its fields. Your Check will be very very specific you need to be accurate to apply it to all code, there might be bunch of false positives.
